aten/src/ATen/native/LegacyBridge.cpp

```diff
@@ -0,0 +1,469 @@
+    return th_norm(self, p);
+  }
+}
+
+Tensor clone(const Tensor& self) {
+  if (_has_native(self)) {
+    return native_clone(self);
+  } else {
+    return th_clone(self);
+  }
+}
+
+Tensor& resize_as_(Tensor& self, const Tensor& the_template) {
+  if (_has_native(self)) {
+    return native_resize_as_(self, the_template);
+  } else {
+    return th_resize_as_(self, the_template);
+  }
+}
+
+Tensor& pow_out(Tensor& result, const Tensor& self, Scalar exponent) {
+  if (_has_native(self)) {
+    return native_pow_out(result, self, exponent);
+  } else {
+    return th_pow_out(result, self, exponent);
+  }
+}
+
+Tensor pow(const Tensor& self, Scalar exponent) {
+  if (_has_native(self)) {
+    return native_pow(self, exponent);
+  } else {
+    return th_pow(self, exponent);
+  }
+}
+
+Tensor& zero_(Tensor& self) {
+  if (_has_native(self)) {
+    return native_zero_(self);
+  } else {
+    return th_zero_(self);
+  }
+}
+
+// Note [CPU sparse is globally native]
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// The current state of affairs is as follows:
+//
+//  - CPU sparse functionality is implemented natively
+//  - CUDA sparse functionality, and all other functionality, are implemented
+//    in TH.
+//
+// Thus, we need these trampoline functions, to help us decide whether or
+// not we can go to native implementations or not.  We expect the trampolines
+// to go away when things get ported to native for real.
+
+// Note [Multiple dispatch to sparse]
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// In principle, there are some degrees of freedom in how we could gotten to our
+// native CPU sparse implementations.  In particular, the handling of how we are
+// simulating multiple dispatch is asymmetric with how th_add_out handles the
+// multiple dispatch.  We justify this in two ways: (1) knowing when to
+// correctly dispatch to sparse requires us to know how what overloads; so we're
+// less likely to get it "wrong" if we put it all together, and (2) all of this
+// is temporary, because we're aiming to natively support multiple dispatch in
+// our system.
+
+Tensor& add_out(Tensor& result, const Tensor& self, const Tensor& other, Scalar alpha) {
+  if (!self.is_cuda()) {
+    // See Note [CPU sparse is globally native] and Note [Multiple dispatch to sparse]
+    auto self_sparse = self.is_sparse();
+    auto other_sparse = other.is_sparse();
+    if (self_sparse && other_sparse) {
+      Tensor b_self, b_other;
+      std::tie(b_self, b_other) = expand_outplace(self, other, "add_out");
+      return s_native_add_out(result, b_self, b_other, alpha);
+    } else if (!self_sparse && other_sparse) {
+      // TODO: Perhaps doing overload selection with SparseTensorRef is
+      // confusing, and we should have given these overloads different names.
+      // For now, we do it this way for consistency with the TH bindings
+      // (not that it is terribly consistent anyway).
+      return native_add_out(result, self, SparseTensorRef(other), alpha);
```

Comment:
I'll admit I don't understand why this calls `native_add_out` but the above calls `s_native_add_out` -- the SparseTensorRef stuff makes sense though.

Reply:
It's pretty simple, actually; in the old Declarations.cwrap, there was no broadcasting enabled for dense-sparse addition, so I mindlessly copied the behavior over.